lib/ts/utils.ts

```diff
@@ -428,3 +428,18 @@ export function toCamelCase(str: string): string {
         return match.toUpperCase().replace("-", "").replace("_", "");
     });
 }
+
+export function toSnakeCase(str: string): string {
+    return str.replace(/[A-Z]/g, (letter) => `_${letter.toLowerCase()}`);
+}
+
+// Transforms the keys of an object from camelCase to snakeCase or vice versa.
+export function transformObjectKeys<T>(obj: { [key: string]: any }, caseType: "snake-case" | "camelCase"): T {
```

Comment:
Are there cases where the objects passed to this have embedded objects (not user-defined)? If so you may want to make this recursive. Not sure if we want to keep these helpers in the long run though.

Reply:
We don't need to make it recursive because only `jwks` and `metadata` fields accept objects and we don't want to alter them.